packages/flutter_test/lib/src/mock_canvas.dart

```diff
@@ -46,6 +46,10 @@ import 'test_async_utils.dart';
 /// paints, among other things, two circles of radius 10.0 and 20.0 (in that
 /// order).
 ///
+/// This matcher does not work well when [Paint] instances are mutated and
+/// reused. Create a new instance of [Paint] for subsequent draw calls instead
+/// of mutating and reusing one instance.
```

Comment:
reusing the paint is fine, but it is just our TestRecordingCanvas can't handle it. I think we should fix the TestRecordingCanvas instead.

Reply:
Good point, that should be doable. I'll take a look.